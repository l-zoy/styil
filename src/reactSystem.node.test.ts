/**
 * @jest-environment node
 */
import {
  createSystem,
  createExtracts as defaultCreateExtracts,
  styled as defaultStyled,
  flush as defaultFlush
} from './reactSystem'

describe('css node', () => {
  it('createStyils node', () => {
    const { createExtracts, styled, flush } = createSystem()

    styled('div', {
      width: 100
    })

    styled('div', {
      width: 100
    })

    styled('div', {
      height: 100
    })

    expect(createExtracts().extractHtml).toMatchSnapshot()

    flush()
  })

  it('node default createExtracts ', () => {
    defaultStyled('div', {
      width: 100
    })

    defaultStyled('div', {
      width: 100
    })

    defaultStyled('div', {
      height: 100
    })

    expect(defaultCreateExtracts().extractHtml).toMatchSnapshot()

    defaultFlush()
  })
})
