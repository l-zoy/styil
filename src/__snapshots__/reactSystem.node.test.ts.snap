// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css node createStyil node 1`] = `
"<meta id="styils-css-cache" name="styils-cache" mode="none" content="css-1473887168|css-2950234073"><style data-ssr="styils-css-ssr">.css-1473887168{width:100px;}</style>
<style data-ssr="styils-css-ssr">.css-2950234073{height:100px;}</style>"
`;

exports[`css node node default createExtracts  1`] = `
"<meta id="styils-css-cache" name="styils-cache" mode="none" content="css-1473887168|css-2950234073"><style data-ssr="styils-css-ssr">.css-1473887168{width:100px;}</style>
<style data-ssr="styils-css-ssr">.css-2950234073{height:100px;}</style>"
`;
